app/assets/javascripts/beak/widgets/ractives/code-editor.coffee

```diff
@@ -49,9 +50,48 @@ window.RactiveModelCodeComponent = Ractive.extend({
     $('#procedurenames-dropdown').trigger('chosen:updated')
     return
 
+  setupAutoComplete: (hintList) ->
+    CodeMirror.registerHelper('hintWords', 'netlogo', hintList)
+    editor = @findComponent('codeEditor').getEditor()
+    editor.on('keyup', (cm, event) =>
+      if not cm.state.completionActive and event.keyCode > 64 and event.keyCode < 91 and @get('autoCompleteStatus')
+        cm.showHint({completeSingle: false})
+    )
+    return
+
+  netLogoHintHelper: (cm, options) ->
+    cur = cm.getCursor()
+    token = cm.getTokenAt(cur)
+    to = CodeMirror.Pos(cur.line, token.end)
+    if token.string and /\S/.test(token.string[token.string.length - 1])
+      term = token.string
+      from = CodeMirror.Pos(cur.line, token.start)
+    else
+      term = ''
+      from = to
+    found = options.words.filter( (word) ->
+      if word.slice(0, term.length) is term
+        word
+    )
+    if found.length > 0
+      return { list: found, from: from, to: to }
+
+  autoCompleteWords: ->
+    autoCompleteWords = Object.keys(@getProcedureNames())
+    keywords = new Set(window.keywords.all)
+    keywords = Array.from(keywords).map((item) -> item.replace("\\", ""))
```

Comment:
I think it makes for confusing code to redeclare variables, especially right after declaring them, especially when the two values are of different types (i.e. `Set` and `Array`).  Please rename one of the`keywords` or inline the `Set` value in the exactly one place where it is referred to.